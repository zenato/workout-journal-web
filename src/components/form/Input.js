import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { formatDate } from 'lib/date';

const Wrapper = styled.input`
  display: block;
  background-color: #fff;
  color: #4a4a4a;
  border: solid #d4d4d4 1px;
  padding: 10px;
  width: 100%;
  text-align: ${({ type }) => (type === 'number' ? 'right' : 'left')};
`;

const Error = styled.span`
  color: #f00;
  font-size: 0.7rem;
  padding: 5px;
`;

const Input = ({ input, label, type, meta: { touched, error } }) => (
  <div>
    <Wrapper
      {...input}
      value={type === 'date' ? formatDate(input.value) : input.value}
      placeholder={label}
      type={type}
    />
    {touched && error && <Error>{error}</Error>}
  </div>
);

Input.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string,
  input: PropTypes.object.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired,
};

export default Input;
