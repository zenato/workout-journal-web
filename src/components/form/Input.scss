@import 'common';

.error {
  color: #f00;
  font-size: 0.7rem;
  padding: 5px;
}
