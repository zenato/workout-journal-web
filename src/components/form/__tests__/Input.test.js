import React from 'react';
import { shallow } from 'enzyme';
import Input from '../Input';

it('renders error', () => {
  const wrapper = shallow(<Input type="text"  value="input" />);
  expect(wrapper).toMatchSnapshot();
});
