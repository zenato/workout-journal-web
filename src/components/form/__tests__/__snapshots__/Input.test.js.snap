// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <input
    className="form-control"
    type="text"
    value="input"
/>,
  "nodes": Array [
    <input
      className="form-control"
      type="text"
      value="input"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Input
        type="text"
        value="input"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "type": "text",
          "value": "input",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <input
          className="form-control"
          type="text"
          value="input"
/>,
        "_debugID": 2,
        "_renderedOutput": <input
          className="form-control"
          type="text"
          value="input"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Input
    type="text"
    value="input"
/>,
}
`;
